### DISCUSSION TITLE
using collection route API for mdx

### DISCUSSION THREAD

**Discussion by hotfudgecake** [gatsbyjs/gatsby#31908] [Help]
Hi everyone! I'm trying to use the collection route API for mdx files.

I've loaded `gatsby-plugin-mdx`, and added a `{Mdx.parent__(File)__name}.js` under `pages`

however, i'm getting this warning 
```
The GraphQL query in the non-page component "../{Mdx.parent__(File)__name}.js" will not be run.
Exported queries are only executed for Page components. It's possible you're
trying to create pages in your gatsby-node.js and that's failing for some
reason.
```

when navigating to the mdx page in browser, it's also not rendered in the layout defined in `{Mdx.parent__(File)__name}.js`.

is there anything else i need to do?

**Comment by KyleAMathews** [comment]
Can you make other collection routes work?
